쿠버네티스 인 액션 - Volume 학습

hotkimho opened the pull request:
- 볼륨 개념, 유형, 사용 예시
- 퍼시스턴트 볼륨
- 동적 프로비저닝

Commit:
docs: 쿠버네티스 volume

hotkimho committed:
docs: 동적 프로비저닝 보충

devmjkim reviewed (commented):
실습 결과가 전혀 없네요.
오탈자 및 충돌 확인해주세요.

hotkimho commented:
> 실습 결과가 전혀 없네요. 오탈자 및 충돌 확인해주세요.

내용 보충 및 실습 결과 추가하고, 오탈자, 깃 충돌부분은 수정하겠습니다.
죄송합니다.

hotkimho committed:
"docs: 볼륨의 필요성 내용 수정"

볼륨의 필요성 내용을 수정했습니다.

hotkimho committed:
docs: 볼륨 유형 추가

볼륨 유형별 설명을 추가 했습니다.

hotkimho committed:
docs:  emptyDir 내용 보충

emptyDir 예제 추가, 각 필드 설명을 추가 했습니다.

hotkimho committed:
docs: hostPath 내용 보충

hostPath 내용 보충 및 실습 예제 추가

hotkimho committed:
docs: nfs 내용 추가

nfs 설명 보충 및 실습 예제 추가

hotkimho committed:
docs: 퍼시스턴트 볼륨 내용 보충

퍼시스턴트 개념 및 실습 예제 추가

hotkimho committed:
docs: 동적 프로비저닝 내용 보충

동적 프로비저닝 보충

hotkimho commented:
2024.06.25 리뷰 보충 내용
- Empydir 메모리 사용 용도
- 프로비저닝 접근 모드 예제 테스트
- 동적프로비넝 테스트

hotkimho committed:
docs: emptyDir memory, disk 기반 설명 보충

emptydir memory, disk 기반에 대한 설명을 보충했습니다.

hotkimho committed:
docs: 퍼시스턴트 볼륨 accessModes 실습 추가

퍼시스턴스 볼륨 accessModes(ReadWriteOncePod) 실습 예제를 추가 했습니다.
ReadWriteOnce, ReadOnlyMany 실습이 원할하지 않아 질문을 남겼습니다.

hotkimho committed:
docs: 동적 프로비저닝 실습 예제 추가

아코디언 프로비지너를 사용하여 동적으로 PV를 생성하는 예제를 추가 했습니다.

hotkimho committed:
docs: 보기 불편한 내용을 보기 쉽게 수정했습니다.

블록 내용, 특수 문자 삭제로 보기 내용을 보기 쉽게 수정했습니다.

hotkimho commented:
### 리뷰 보충한 내용
- emptyDir 메모리, 디스크 기반 볼륨 정리
- PV accessModes 실습
- 동적 프로비저닝(아코디언 프로비지너 사용)

### 궁금한 내용
PV accessModes를 실습하며 궁금한 내용이 있습니다.
쿠버네티스 공식문서에
```
쿠버네티스는 볼륨 접근 모드를 이용해 퍼시스턴트볼륨클레임과 퍼시스턴트볼륨을 연결한다.
경우에 따라 볼륨 접근 모드는 퍼시스턴트볼륨을 탑재할 수 있는 위치도 제한한다.
볼륨 접근 모드는 스토리지를 마운트 한 후에는 쓰기 보호를 적용하지 않는다. 
접근 모드가 ReadWriteOnce, ReadOnlyMany 혹은 ReadWriteMany로 지정된 경우에도 접근 모드는 볼륨에 제약 조건을 설정하지 않는다.
예를 들어 퍼시스턴트볼륨이 ReadOnlyMany로 생성되었다 하더라도, 해당 퍼시스턴트 볼륨이 읽기 전용이라는 것을 보장하지 않는다.
만약 접근 모드가 ReadWriteOncePod로 지정된 경우, 볼륨에 제한이 설정되어 단일 파드에만 마운트 할 수 있게 된다.
```
내용과 함께 표 데이터를 보면 NFS의 경우 (ReadWriteOnce, ReadOnlyMany, ReadWriteMany) 설정들이 의도에 맞게 보장되지 않는다는 내용으로 명시가 되어 있습니다.

의도(ReadWriteOnce, ReadOnlyMany mode)에 맞게 동작하려면 어떻게 설정해야 하는지 궁금합니다.
 
@devmjkim

devmjkim commented:
> ### 리뷰 보충한 내용
> * emptyDir 메모리, 디스크 기반 볼륨 정리
> * PV accessModes 실습
> * 동적 프로비저닝(아코디언 프로비지너 사용)
> 
> ### 궁금한 내용
> PV accessModes를 실습하며 궁금한 내용이 있습니다. 쿠버네티스 공식문서에
> 
> ```
> 쿠버네티스는 볼륨 접근 모드를 이용해 퍼시스턴트볼륨클레임과 퍼시스턴트볼륨을 연결한다.
> 경우에 따라 볼륨 접근 모드는 퍼시스턴트볼륨을 탑재할 수 있는 위치도 제한한다.
> 볼륨 접근 모드는 스토리지를 마운트 한 후에는 쓰기 보호를 적용하지 않는다. 
> 접근 모드가 ReadWriteOnce, ReadOnlyMany 혹은 ReadWriteMany로 지정된 경우에도 접근 모드는 볼륨에 제약 조건을 설정하지 않는다.
> 예를 들어 퍼시스턴트볼륨이 ReadOnlyMany로 생성되었다 하더라도, 해당 퍼시스턴트 볼륨이 읽기 전용이라는 것을 보장하지 않는다.
> 만약 접근 모드가 ReadWriteOncePod로 지정된 경우, 볼륨에 제한이 설정되어 단일 파드에만 마운트 할 수 있게 된다.
> ```
> 
> 내용과 함께 표 데이터를 보면 NFS의 경우 (ReadWriteOnce, ReadOnlyMany, ReadWriteMany) 설정들이 의도에 맞게 보장되지 않는다는 내용으로 명시가 되어 있습니다.
> 
> 의도(ReadWriteOnce, ReadOnlyMany mode)에 맞게 동작하려면 어떻게 설정해야 하는지 궁금합니다.
> 
> @devmjkim

아래 stackoverflow와 비슷한 질문이네요. nfs나 애플리케이션 단에서 막아야한다고 하네요. 그리고 nfs의 경우 공유 스토리지라 ReadWriteMany / ReadOnlyMany가 일반적이라고하니 아래 링크 참고해주세요. 
https://stackoverflow.com/questions/77689703/does-nfs-server-as-a-pv-respect-the-access-mode-in-kubernetes-cluster

hotkimho commented:
> > ### 리뷰 보충한 내용
> > 
> > * emptyDir 메모리, 디스크 기반 볼륨 정리
> > * PV accessModes 실습
> > * 동적 프로비저닝(아코디언 프로비지너 사용)
> > 
> > ### 궁금한 내용
> > PV accessModes를 실습하며 궁금한 내용이 있습니다. 쿠버네티스 공식문서에
> > ```
> > 쿠버네티스는 볼륨 접근 모드를 이용해 퍼시스턴트볼륨클레임과 퍼시스턴트볼륨을 연결한다.
> > 경우에 따라 볼륨 접근 모드는 퍼시스턴트볼륨을 탑재할 수 있는 위치도 제한한다.
> > 볼륨 접근 모드는 스토리지를 마운트 한 후에는 쓰기 보호를 적용하지 않는다. 
> > 접근 모드가 ReadWriteOnce, ReadOnlyMany 혹은 ReadWriteMany로 지정된 경우에도 접근 모드는 볼륨에 제약 조건을 설정하지 않는다.
> > 예를 들어 퍼시스턴트볼륨이 ReadOnlyMany로 생성되었다 하더라도, 해당 퍼시스턴트 볼륨이 읽기 전용이라는 것을 보장하지 않는다.
> > 만약 접근 모드가 ReadWriteOncePod로 지정된 경우, 볼륨에 제한이 설정되어 단일 파드에만 마운트 할 수 있게 된다.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 내용과 함께 표 데이터를 보면 NFS의 경우 (ReadWriteOnce, ReadOnlyMany, ReadWriteMany) 설정들이 의도에 맞게 보장되지 않는다는 내용으로 명시가 되어 있습니다.
> > 의도(ReadWriteOnce, ReadOnlyMany mode)에 맞게 동작하려면 어떻게 설정해야 하는지 궁금합니다.
> > @devmjkim
> 
> 아래 stackoverflow와 비슷한 질문이네요. nfs나 애플리케이션 단에서 막아야한다고 하네요. 그리고 nfs의 경우 공유 스토리지라 ReadWriteMany / ReadOnlyMany가 일반적이라고하니 아래 링크 참고해주세요. https://stackoverflow.com/questions/77689703/does-nfs-server-as-a-pv-respect-the-access-mode-in-kubernetes-cluster

내용확인했습니다.
링크 공유해주셔서 감사합니다.
